Rebase Kubernetes

kargakis opened the pull request:
@pweil- @deads2k @smarterclayton @liggitt 

### TODO:
- [x] bump Kube (86b4e777e1947c1bc00e422306a3ca74cbd54dbe)
- [x] restore incorrectly backleveled packages
- [x] squash not shared bumps into the Kube bump
- [x] cherry pick unmerged UPSTREAM commits
- [x] cherry-pick https://github.com/kubernetes/kubernetes/pull/13864
- [x] cherry-pick https://github.com/kubernetes/kubernetes/pull/13910
- [x] ~~fix for https://github.com/kubernetes/kubernetes/issues/13812~~ carry a patch that disables schema validation in the client
- [x] necessary Origin refactoring
- [x] unit tests pass
- [x] test-cmd.sh passes
- [x] integration tests pass
- [x] e2e tests pass
- [x] verify-open-ports.sh passes

Should readd https://github.com/openshift/origin/commit/f0aeb08de5c3b2938e014d26b33ba1d1cd6a0e24 if docker-end-to-end fails for mount/nsenter

### Wall of shame:
* https://github.com/openshift/origin/commit/cb1e86c014ba93d5c3eb7cdc19258a371988df63: UPSTREAM: 8890: Allowing ActiveDeadlineSeconds to be updated for a pod
* https://github.com/openshift/origin/commit/87cead531d4bea61313c1f37234f0e0767512e4d: UPSTREAM: 7893: scc
* https://github.com/openshift/origin/commit/7ded3bc1faeee43353901bf6e1287694bcf1f78e: UPSTREAM: 7893: scc allocation interface methods
*  https://github.com/kargakis/origin/commit/48b07cd4bf28a6522247f2f1d00d03864a3e73d8: UPSTREAM: 9870: Allow Recyclers to be configurable (Updated as per the latest changes in the upstream PR)
* https://github.com/openshift/origin/commit/787f6893162c8132394f60073572f3280f14dc01: UPSTREAM: 9844: EmptyDir volume SELinux support
*  https://github.com/openshift/origin/commit/87a0ec7d3e12d15ff8b404ad74c532c1fa6c9a7d: UPSTREAM: 9009: Retry service account update when adding token reference
* https://github.com/openshift/origin/commit/c826b0ea264ecceaa5bd8eda5b85a4802e15d9c9: UPSTREAM: 12498: Re-add timeouts for kubelet
* https://github.com/openshift/origin/commit/223994f2dd24dc3806e67c2aa57fefa84bd1c244:  UPSTREAM: 12221: Allow custom namespace creation in e2e framework 
* https://github.com/openshift/origin/commit/2bc193a24614dc15a3257f7953dbb5a02dd9dc1a:  UPSTREAM: 11827: allow permissive SA secret ref limitting 
* https://github.com/openshift/origin/commit/2353ad26c474594bd8c62ced6afad54f772b3b84:  UPSTREAM: 13746: Fix field=metadata.name 

Next rebase will pick these up as they are already merged:
* https://github.com/openshift/origin/commit/c36fc44d4aecc4fd398c9c47dc165ee65fb49a07:  UPSTREAM: 13756: expose: Avoid selector resolution if a selector is not needed
* https://github.com/openshift/origin/commit/32ce7f27691099ffbeee451fb74ca2de7c36db5a:  UPSTREAM: 14063: enable system CAs
* https://github.com/kargakis/origin/commit/9dcef68409e1c94c9b99d400ca38ea796cbe613b: UPSTREAM: 13864: Fix kubelet logs --follow bug
* https://github.com/kargakis/origin/commit/08ea22a43368ffc5986dbc4f32df504a851d8218: UPSTREAM: 13910: Fix resourcVersion = 0 in cacher
* https://github.com/openshift/origin/commit/fb148cb82490c4430c873196e2f009351b613927: UPSTREAM: 14291: add patch verb to APIRequestInfo
* https://github.com/openshift/origin/commit/e3cb199a7ea72fb6d0519aa2fd751b68395a700e:  UPSTREAM: 13728: Allow to replace os.Exit() with panic when CLI command fatal
* https://github.com/kargakis/origin/commit/24456364b19526a5cd24ba1e3c9c935e621a45ef: UPSTREAM: 14502: don't fatal on missing sorting flag
* https://github.com/kargakis/origin/commit/7b6f8c97d8d4c326ddace5137ddd2df0314353e4: UPSTREAM: 14664: fix testclient prepend

### Post-rebase issues:
- https://github.com/openshift/origin/issues/4675 (for https://github.com/openshift/origin/commit/de466c41217f13ce8837cdcb6942d3094dcfc779:  UPSTREAM: vjeantet/ldapserver: 15: fix ldapserver test panic)
- https://github.com/openshift/origin/issues/4781
- https://github.com/openshift/origin/issues/4785
- https://github.com/openshift/origin/issues/4796
- https://github.com/openshift/origin/issues/4797
- https://github.com/openshift/origin/issues/4799

Closes #4434

deads2k commented:
Regarding https://github.com/kubernetes/kubernetes/pull/9870, I think you should re-pick it.  There has been some action on that pull.

kargakis commented:
> Regarding kubernetes/kubernetes#9870, I think you should re-pick it. There has been some action on that pull.

Just pushed the newest changes in

deads2k commented:
https://github.com/openshift/origin/commit/81a137bf19d6cc93e714773a81b33f684faf8ab0 has been fixed upstream.

I thought we had an issue for @fabianofranz  to fix up https://github.com/openshift/origin/commit/199adb7fd03ad0375ca9a7e8fa31aec9f372ab8b, but until then, you should carry it.

deads2k commented:
There's a script that @pmorie wrote to pull in the dependencies for the kube unit tests.  See https://github.com/openshift/origin/blob/master/hack/copy-kube-artifacts.sh. That should be used to create a new patch instead of carrying https://github.com/openshift/origin/commit/b8f66b9f1453822d86c6ed89f83320dfb36998d8.

@smarterclayton you're more familiar with `git` than I am.  Should we start squashing that unit test patch into the `bump` commit to limit "create", "delete", "create" noise and size in our git repo?

liggitt commented:
> Should we start squashing that unit test patch into the bump commit to limit "create", "delete", "create" noise and size in our git repo?

Yes please

liggitt commented:
> c826b0e: Upstream they decided not to take kubernetes/kubernetes#12498, should we carry or drop ours?

I still have a todo to prove that dropping the timeout can leave hung connections from the master to the kubelet. We should carry until then.

kargakis commented:
Added back https://github.com/openshift/origin/commit/199adb7fd03ad0375ca9a7e8fa31aec9f372ab8b and https://github.com/openshift/origin/commit/c826b0ea264ecceaa5bd8eda5b85a4802e15d9c9

fabianofranz commented:
I have https://github.com/openshift/origin/issues/4172, but it has not yet
been fixed. Please carry it for now.


On Tue, Sep 15, 2015 at 10:03 AM, David Eads <notifications@github.com>
wrote:

> 81a137b
> <https://github.com/openshift/origin/commit/81a137bf19d6cc93e714773a81b33f684faf8ab0>
> has been fixed upstream.
>
> I thought we had an issue for @fabianofranz
> <https://github.com/fabianofranz> to fix up 199adb7
> <https://github.com/openshift/origin/commit/199adb7fd03ad0375ca9a7e8fa31aec9f372ab8b>,
> but until then, you should carry it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4663#issuecomment-140381066>.
>

deads2k commented:
be sure to re-pick https://github.com/openshift/origin/pull/4703

deads2k commented:
About all those bumps.  
 1. I've been going with the rule that if its is *only* a transitive dependency, then I roll it into the make kube bump.  If it is a shared dependency, then separate it out to make it obvious.  That means something like "blackfriday" is combined into the kube commit, but "go-bindata-assetfs" is separate.
 1. Every bump needs to be evaluated to make sure we weren't carrying patches, if so you'll need to re-apply the patches as well.
 1. Before you bump the packages, you should check to make sure that you're backleveling us.  Many times, the commits do not agree and kube is running behind.  Taking the bump can inadvertently backlevel our dependencies.  Open issues for those.

liggitt commented on Godeps/Godeps.json:
```diff
@@ -373,12 +393,12 @@
 		{
 			"ImportPath": "github.com/elazarl/go-bindata-assetfs",
-			"Rev": "3dcc96556217539f50599357fb481ac0dc7439b9"
+			"Rev": "c57a80f1ab2ad67bafa83f5fd0b4c2ecbd253dd5"
```
We want our version, not kube's. https://github.com/kubernetes/kubernetes/pull/13835 Is open to bring kube up to our level

kargakis replied:
Thanks for this, godep save also backleveled logrus and golang.org/net but I guess we want our versions, especially for the net package.

liggitt commented on Godeps/Godeps.json:
```diff
@@ -645,72 +665,57 @@
-			"Comment": "v0.1-164-g9d342eb",
-			"Rev": "0f9e6558e8dceb8c8317e3587d9c9c94ae07ecb8"
+			"Comment": "v0.1-196-ge6604ec",
+			"Rev": "e6604ec1114b1141c735114c08dc717a9a717929"
```
Rebase shouldn't be changing sdn imports, should it?

liggitt commented on Godeps/Godeps.json:
```diff
@@ -645,72 +665,57 @@
-			"Comment": "v0.1-164-g9d342eb",
-			"Rev": "0f9e6558e8dceb8c8317e3587d9c9c94ae07ecb8"
+			"Comment": "v0.1-196-ge6604ec",
+			"Rev": "e6604ec1114b1141c735114c08dc717a9a717929"
```
Same here

liggitt commented:
Additional todos:
* run extended tests
* manually run https://github.com/openshift/origin/blob/master/hack/verify-open-ports.sh to ensure no additional ports are open

ncdc commented:
What about the webhook commits? We have 1 to carry, followed by a revert.

liggitt commented:
Possible cherry-picks:
* https://github.com/kubernetes/kubernetes/pull/13864 - fix `logs --follow` bug?
* https://github.com/kubernetes/kubernetes/issues/13812 eventual fix
* https://github.com/kubernetes/kubernetes/issues/13809 - fix watch ?resourceVersion=0

kargakis commented:
> About all those bumps.

I separated the bumps mainly for easier reviewing. I'll squash back those coming from Kube at some point in the end of this.

kargakis commented:
> What about the webhook commits? We have 1 to carry, followed by a revert.

I already have added those.

kargakis commented:
Turns out there were a few backleveled packages I had to revert:

* golang.org/x/net
* github.com/gogo/protobuf/proto
* github.com/coreos/go-semver/semver
* github.com/miekg/dns
* github.com/Sirupsen/logrus

ncdc commented:
I am suggesting / wondering if we even need them any more.

On Friday, September 18, 2015, Michail Kargakis <notifications@github.com>
wrote:

> What about the webhook commits? We have 1 to carry, followed by a revert.
>
> I already have added those.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4663#issuecomment-141373660>.
>

kargakis commented:
> I am suggesting / wondering if we even need them any more.

Ah, cool, I was thinking of removing them entirely but then we may want to revert the reverted commit at some point? Not sure about it @smarterclayton ?

kargakis commented:
> be sure to re-pick #4703

Hmm, it seems that I have already squashed it into the bump commit. Since it is already merged upstream is it ok to leave it as is or should I add it back as an UPSTREAM commit?

ncdc commented:
I believe we moved the code into pkg and don't plan on putting it upstream.
Clayton can confirm.

On Friday, September 18, 2015, Michail Kargakis <notifications@github.com>
wrote:

> I am suggesting / wondering if we even need them any more.
>
> Ah, cool, I was thinking of removing them entirely but then we may want to
> revert the reverted commit at some point? Not sure about it
> @smarterclayton <https://github.com/smarterclayton> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4663#issuecomment-141398167>.
>

kargakis commented:
> Hmm, it seems that I have already squashed it into the bump commit. Since it is already merged upstream is it ok to leave it as is or should I add it back as an UPSTREAM commit?

Ok, added this as an UPSTREAM commit

kargakis commented:
@deads2k any idea on this?
```
# k8s.io/kubernetes/pkg/api/v1beta3
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:733: undefined: api.MetadataFile
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:735: undefined: api.MetadataFile
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:744: undefined: api.MetadataVolumeSource
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:746: undefined: api.MetadataVolumeSource
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:2249: in.Metadata undefined (type *api.VolumeSource has no field or method Metadata)
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:2251: in.Metadata undefined (type *api.VolumeSource has no field or method Metadata)
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:2966: undefined: api.MetadataFile
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:2977: undefined: api.MetadataVolumeSource
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:2982: undefined: api.MetadataFile
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:4483: out.Metadata undefined (type *api.VolumeSource has no field or method Metadata)
Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion_generated.go:4483: too many errors
Makefile:31: recipe for target 'build' failed
make: *** [build] Error 1
```

I see MetadataFile defined only in v1beta3, but that's exactly the same commit from your rebase.

deads2k commented:
> I see MetadataFile defined only in v1beta3, but that's exactly the same commit from your rebase.

That came in as part of 5093.  Looks like the type name changed before the pull was merged.  That's a bit sticky.

@smarterclayton how backwards compatible is backwards compatible?

smarterclayton commented:
web hooks are gone, you can drop both commits.

This is something that @pmorie needs to assess - who was using the early
metadata volume (v3 folks?) and if so we have to preserve backcompat
somehow.

On Fri, Sep 18, 2015 at 7:40 AM, David Eads <notifications@github.com>
wrote:

> I see MetadataFile defined only in v1beta3, but that's exactly the same
> commit from your rebase.
>
> That came in as part of 5093. Looks like the type name changed before the
> pull was merged. That's a bit sticky.
>
> @smarterclayton <https://github.com/smarterclayton> how backwards
> compatible is backwards compatible?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4663#issuecomment-141424988>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift

pmorie commented:
@smarterclayton I think we might be able to have a conversion that handles the metadata volume.

pmorie commented:
@deads2k @smarterclayton We could also carry a patch that adds metadata to the volume sources and adds a plugin that wraps the downward API plugin.

deads2k commented:
> @deads2k @smarterclayton We could also carry a patch that adds metadata to the volume sources and adds a plugin that wraps the downward API plugin.

I was rather hoping to avoid that, but if it is in use, it would be pretty atrocious to suddenly have your pods fail and then the fields dropped (unknown fields are removed), so that even a rollback won't solve your problem without restoring from backups.

I think if we have to preserve the functionality, we should keep the types and change them to the new field types during conversion.  Going to the external, we always make both match based on the new field.  Going to internal, we use the old field to stomp the new field if present (old clients will drop the new field, current clients will have force matched fields, and new clients will drop the old field once its gone).

pmorie commented:
@deads2k @smarterclayton

@sdminonne Tells me Amadeus isn't using the old plugin yet, and that from their standpoint we can nuke it as long as the upstream downward API plugin is available.

kargakis commented:
@deads2k updated v1beta3 to use the new fields and split Origin commits into interesting and boring (pretty sure there is more refactoring tbd).

smarterclayton commented:
Where are we overall on this rebase?

On Sep 21, 2015, at 9:52 AM, Michail Kargakis <notifications@github.com>
wrote:

@deads2k <https://github.com/deads2k> updated v1beta3 to use the new fields
and split Origin commits into interesting and boring (pretty sure there is
more refactoring tbd).

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4663#issuecomment-141985634>.

smarterclayton commented:
I'm not sure we can simply drop plugins for a running system, regardless of
whether we know they aren't in use.  We can make it be deprecated for 1.1
maybe, but not removed.

On Sep 21, 2015, at 9:45 AM, Paul Morie <notifications@github.com> wrote:

@deads2k <https://github.com/deads2k> @smarterclayton
<https://github.com/smarterclayton>

@sdminonne <https://github.com/sdminonne> Tells me Amadeus isn't using the
old plugin yet, and that from their standpoint we can nuke it as long as
the upstream downward API plugin is available.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4663#issuecomment-141982756>.

smarterclayton commented:
It's possible we labeled it as experimental, but I doubt it.

On Sep 21, 2015, at 9:45 AM, Paul Morie <notifications@github.com> wrote:

@deads2k <https://github.com/deads2k> @smarterclayton
<https://github.com/smarterclayton>

@sdminonne <https://github.com/sdminonne> Tells me Amadeus isn't using the
old plugin yet, and that from their standpoint we can nuke it as long as
the upstream downward API plugin is available.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4663#issuecomment-141982756>.

kargakis commented:
> Where are we overall on this rebase?

Compiles, need to fix tests.

pmorie commented:
@kargakis, @deads2k and I spent some time talking about metadata volumes today.  I think the right solution is to convert metadata into downward API in the internal API, and then convert downward API back out on the outbound conversion.

deads2k commented:
> @kargakis, @deads2k and I spent some time talking about metadata volumes today. I think the right solution is to convert metadata into downward API in the internal API, and then convert downward API back out on the outbound conversion.

@kargakis This would count as a very interesting change.  Please leave it as a separate commit.  In the past, I've run a checklist in the description for commits I want reviewed by particular people to help keep track of them.

pweil- commented:
> @kargakis, @deads2k and I spent some time talking about metadata volumes today. I think the right solution is to convert metadata into downward API in the internal API, and then convert downward API back out on the outbound conversion.

I'm interested in what the solution is here since I am in a similar situation with SCC.

kargakis commented:
Unit test failures boiled down to:
```
--- FAIL: TestValidateImageStream (0.00s)
	validation_test.go:331: missing name: unexpected errors: [{"Type":"FieldValueRequired","Field":"metadata.name","BadValue":"","Detail":"
		
		A: "}]
		
		B: name or generateName is required"}]
		
FAIL
FAIL	github.com/openshift/origin/pkg/image/api/validation	0.021s
```
```
--- FAIL: TestDeepCopyApiObjects (0.26s)
	copy_test.go:46: 
FAIL
FAIL	github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/api	1.977s
```
```
--- FAIL: TestExampleObjectSchemas (0.12s)
	examples_test.go:351: skipping : ../examples/elasticsearch/apiserver-secret.yaml/apiserver-secret
	examples_test.go:346: ../examples/elasticsearch/mytunes-namespace.yaml: mytunes-namespace does not have a test case defined
	examples_test.go:351: skipping : ../examples/openshift-origin/secret.json/secret
	examples_test.go:351: skipping : ../docs/user-guide/multi-pod.yaml/multi-pod
FAIL
FAIL	github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/examples	0.155s
...
--- FAIL: TestInstallThirdPartyAPIGet (0.00s)
	fake_etcd_client.go:201: creating //ThirdPartyResourceData/company.com/foos/default/test, index 0 (ttl: 0)
	fake_etcd_client.go:142: data for /ThirdPartyResourceData/company.com/foos/default/test was not defined prior to invoking Get
        Error Trace:    master_test.go:566
			master_test.go:537
	Error:		Not equal: 200 (expected)
			        != 404 (actual)
		
        Error Trace:    master_test.go:575
			master_test.go:537
	Error:		Should be true
		
	master_test.go:576: expected:
		master.Foo{TypeMeta:api.TypeMeta{Kind:"Foo", APIVersion:"v1"}, ObjectMeta:api.ObjectMeta{Name:"test", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:util.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*util.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil)}, SomeField:"test field", OtherField:10}
		saw:
		master.Foo{TypeMeta:api.TypeMeta{Kind:"Status", APIVersion:"v1"}, ObjectMeta:api.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:util.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*util.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil)}, SomeField:"", OtherField:0}
	fake_etcd_client.go:201: creating //ThirdPartyResourceData/company.com/foos/default/test, index 0 (ttl: 0)
	fake_etcd_client.go:142: data for /ThirdPartyResourceData/company.com/foos/default/test was not defined prior to invoking Get
        Error Trace:    master_test.go:566
			master_test.go:537
	Error:		Not equal: 200 (expected)
			        != 404 (actual)
		
        Error Trace:    master_test.go:575
			master_test.go:537
	Error:		Should be true
		
	master_test.go:576: expected:
		master.Foo{TypeMeta:api.TypeMeta{Kind:"Foo", APIVersion:"v3"}, ObjectMeta:api.ObjectMeta{Name:"test", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:util.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*util.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil)}, SomeField:"test field", OtherField:10}
		saw:
		master.Foo{TypeMeta:api.TypeMeta{Kind:"Status", APIVersion:"v1"}, ObjectMeta:api.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:util.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*util.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil)}, SomeField:"", OtherField:0}
--- FAIL: TestInstallThirdPartyAPIPost (0.00s)
	fake_etcd_client.go:201: creating /ThirdPartyResourceData/company.com/foos/default/test, index 0 (ttl: 0)
	fake_etcd_client.go:142: data for //ThirdPartyResourceData/company.com/foos/default/test was not defined prior to invoking Get
        Error Trace:    master_test.go:627
			master_test.go:582
	Error:		No error is expected but got 100:  () [0]
		
--- FAIL: TestInstallThirdPartyAPIDelete (0.01s)
	fake_etcd_client.go:201: creating //ThirdPartyResourceData/company.com/foos/default/test, index 0 (ttl: 0)
	fake_etcd_client.go:142: data for /ThirdPartyResourceData/company.com/foos/default/test was not defined prior to invoking Get
        Error Trace:    master_test.go:677
			master_test.go:648
	Error:		Not equal: 200 (expected)
			        != 404 (actual)
		
        Error Trace:    master_test.go:685
			master_test.go:648
	Error:		Should be true
		
	master_test.go:686: expected:
		{{Foo } {test      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[]} test field 10}
		saw:
		{{Status v1} {      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[]}  0}
	fake_etcd_client.go:142: data for /ThirdPartyResourceData/company.com/foos/default/test was not defined prior to invoking Get
        Error Trace:    master_test.go:694
			master_test.go:648
	Error:		Not equal: 200 (expected)
			        != 404 (actual)
		
	fake_etcd_client.go:142: data for /ThirdPartyResourceData/company.com/foos/default/test was not defined prior to invoking Get
        Error Trace:    master_test.go:705
			master_test.go:648
	Error:		Should be true
		
	master_test.go:706: unexpected deleted keys: []
	fake_etcd_client.go:201: creating //ThirdPartyResourceData/company.com/foos/default/test, index 0 (ttl: 0)
	fake_etcd_client.go:142: data for /ThirdPartyResourceData/company.com/foos/default/test was not defined prior to invoking Get
        Error Trace:    master_test.go:677
			master_test.go:648
	Error:		Not equal: 200 (expected)
			        != 404 (actual)
		
        Error Trace:    master_test.go:685
			master_test.go:648
	Error:		Should be true
		
	master_test.go:686: expected:
		{{Foo } {test      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[]} test field 10}
		saw:
		{{Status v1} {      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[]}  0}
	fake_etcd_client.go:142: data for /ThirdPartyResourceData/company.com/foos/default/test was not defined prior to invoking Get
        Error Trace:    master_test.go:694
			master_test.go:648
	Error:		Not equal: 200 (expected)
			        != 404 (actual)
		
	fake_etcd_client.go:142: data for /ThirdPartyResourceData/company.com/foos/default/test was not defined prior to invoking Get
        Error Trace:    master_test.go:705
			master_test.go:648
	Error:		Should be true
		
	master_test.go:706: unexpected deleted keys: []
FAIL
FAIL	github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/master	3.782s
```

kargakis commented:
https://github.com/kargakis/origin/commit/b8f66b9f1453822d86c6ed89f83320dfb36998d8 wasn't dropped by the rebase so there is no need for running `./hack/copy-kube-artifacts.sh` (except for diffs in the artifacts)

kargakis commented:
Serialization test for DeploymentConfigs fails randomly: http://pastebin.com/AWMKiFbx

@ironcladlou any ideas?

ironcladlou commented:
Not yet, although it looks related to volumes.

Left:

```
EmptyDir: (*api.EmptyDirVolumeSource)(<nil>),
...
Cinder: (*api.CinderVolumeSource)(0xc20824a900)({   
 VolumeID: (string) (len=19) "zVŨÑ#\\9鯒娰鉛.",      
 FSType: (string) (len=35) "ɶǿaǼȍ蟬r4揋 *柮呁ļ#蟠Ĥʚ",
 ReadOnly: (bool) true                              
}),                                                 
```

Right:

```
EmptyDir: (*api.EmptyDirVolumeSource)(0xc20829be60)({
 Medium: (api.StorageMedium) ""
}),
...
Cinder: (*api.CinderVolumeSource)(<nil>),
```

liggitt commented:
did the cinder stuff get stomped in v1beta3 or v1 types or conversions?

kargakis commented:
> did the cinder stuff get stomped in v1beta3 or v1 types or conversions?

Hmm, i didn't add them in v1beta3 and I suppose I have to:)

pweil- commented:
@kargakis looks like they copy test is failing because it is not copying the `MetadataVolumeSource`.  Need to regen those too while you're fixing up the conversions.

debugging:
```
	copy_test.go:56: volumes not equal
	copy_test.go:59: volume 0 is not equal
	copy_test.go:60: a: v1.Volume{Name:"竁ƀț2鶀ɿ鵐<o觚n鞓gS定a凯VȚ", VolumeSource:v1.VolumeSource{HostPath:(*v1.HostPathVolumeSource)(nil), EmptyDir:(*v1.EmptyDirVolumeSource)(0xc20818afe0), GCEPersistentDisk:(*v1.GCEPersistentDiskVolumeSource)(nil), AWSElasticBlockStore:(*v1.AWSElasticBlockStoreVolumeSource)(0xc20816a8d0), GitRepo:(*v1.GitRepoVolumeSource)(0xc20817eb20), Secret:(*v1.SecretVolumeSource)(0xc20818b0c0), NFS:(*v1.NFSVolumeSource)(nil), ISCSI:(*v1.ISCSIVolumeSource)(nil), Glusterfs:(*v1.GlusterfsVolumeSource)(0xc20816a9c0), PersistentVolumeClaim:(*v1.PersistentVolumeClaimVolumeSource)(nil), RBD:(*v1.RBDVolumeSource)(nil), Cinder:(*v1.CinderVolumeSource)(nil), CephFS:(*v1.CephFSVolumeSource)(nil), DownwardAPI:(*v1.DownwardAPIVolumeSource)(0xc20817ec80), Metadata:(*v1.MetadataVolumeSource)(0xc20817ed40)}}
	copy_test.go:61: b: v1.Volume{Name:"竁ƀț2鶀ɿ鵐<o觚n鞓gS定a凯VȚ", VolumeSource:v1.VolumeSource{HostPath:(*v1.HostPathVolumeSource)(nil), EmptyDir:(*v1.EmptyDirVolumeSource)(0xc20818b580), GCEPersistentDisk:(*v1.GCEPersistentDiskVolumeSource)(nil), AWSElasticBlockStore:(*v1.AWSElasticBlockStoreVolumeSource)(0xc20816ae10), GitRepo:(*v1.GitRepoVolumeSource)(0xc20817f2e0), Secret:(*v1.SecretVolumeSource)(0xc20818b590), NFS:(*v1.NFSVolumeSource)(nil), ISCSI:(*v1.ISCSIVolumeSource)(nil), Glusterfs:(*v1.GlusterfsVolumeSource)(0xc20816ae40), PersistentVolumeClaim:(*v1.PersistentVolumeClaimVolumeSource)(nil), RBD:(*v1.RBDVolumeSource)(nil), Cinder:(*v1.CinderVolumeSource)(nil), CephFS:(*v1.CephFSVolumeSource)(nil), DownwardAPI:(*v1.DownwardAPIVolumeSource)(0xc20817f300), Metadata:(*v1.MetadataVolumeSource)(nil)}}
```

kargakis commented:
Plain `./hack/update-generated-deep.copies.sh` and `./hack/update-generated-conversions.sh` won't make the desired updates here so I had to push v1beta3 and scc in a Kube branch (https://github.com/kargakis/kubernetes/tree/rebase-v1beta3). Still v1beta3 needs to account for changes in ServiceSpec and I am not sure what to add/edit/remove there. Anybody interested in updating v1beta3?:) @deads2k @pweil- @smarterclayton

kargakis commented:
@deads2k @pmorie I guess we will have to update validation for volume sources as well right? https://github.com/openshift/origin/blob/1b11e8625a1cfd6071ad300440c737823765f436/Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/validation/validation.go#L373-L379

deads2k commented:
> @deads2k @pmorie I guess we will have to update validation for volume sources as well right? https://github.com/openshift/origin/blob/1b11e8625a1cfd6071ad300440c737823765f436/Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/validation/validation.go#L373-L379

I wasn't expecting the internal objects to include `Metadata`

kargakis commented:
Ah need to update it to use Downward

On Wed, Sep 23, 2015 at 5:52 PM, David Eads <notifications@github.com>
wrote:

> @deads2k <https://github.com/deads2k> @pmorie <https://github.com/pmorie>
> I guess we will have to update validation for volume sources as well right?
> https://github.com/openshift/origin/blob/1b11e8625a1cfd6071ad300440c737823765f436/Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/validation/validation.go#L373-L379
>
> I wasn't expecting the internal objects to include Metadata
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4663#issuecomment-142645971>.
>

kargakis commented:
> Ah need to update it to use Downward

False alarm, this is already updated in the rebased Kube dep.

pweil- commented:
@kargakis I submitted a PR to your branch that fixes the unit tests.  It is:

1.  Adding the necessary types to v1beta3 with regenerated conversions/deep copies from kube for api, api/v1, api/v1beta3, apis/experimental
1.  Using the generated conversion methods in `conversion.go` but adding your custom logic
1.  Renaming the `custom_*` methods to `convert_*` so generated methods will be able to use them
1.  Fixes the ImageStream test that need an error message update

PTAL and make sure the conversion logic is capturing what you need in each api.

kargakis commented:
@pweil- thanks for this. Changes look good. I'll have to break it down to pieces, some of those need to be merged in the v1beta3 carry, others in the commit that takes care of backwards comp and the test fix needs to be with the rest of the origin changes.

kargakis commented:
Travis and `./hack/verify-generated-deep-copies.sh` locally both fail for our deep copies:
```
Diffing current deep copies against freshly generated deep copies...

Files /home/travis/gopath/src/github.com/openshift/origin/pkg/api/deep_copy_generated.go and /home/travis/gopath/src/github.com/openshift/origin/_output/verify-generated-deep-copies/pkg/api/deep_copy_generated.go differ

Files /home/travis/gopath/src/github.com/openshift/origin/pkg/api/v1/deep_copy_generated.go and /home/travis/gopath/src/github.com/openshift/origin/_output/verify-generated-deep-copies/pkg/api/v1/deep_copy_generated.go differ

Files /home/travis/gopath/src/github.com/openshift/origin/pkg/api/v1beta3/deep_copy_generated.go and /home/travis/gopath/src/github.com/openshift/origin/_output/verify-generated-deep-copies/pkg/api/v1beta3/deep_copy_generated.go differ

FAILURE: Current deep copies out of date. Please run hack/update-generated-deep-copies.sh

make: *** [verify] Error 1
```
Funny thing is run `/.hack/update-generated-deep-copies.sh` then I get compiler errors:)
```
[mkargaki@dhcp129-211 origin]$ ./hack/verify-generated-deep-copies.sh 
===== Verifying Generated Conversions =====
Building gendeepcopy binary...
   ++ Building go targets for linux/amd64: cmd/gendeepcopy
   ++ Placing binaries
   Verifying gendeepcopy binary...
Generating fresh deep copies...
Diffing current deep copies against freshly generated deep copies...
Files /home/mkargaki/go/src/github.com/kargakis/origin/pkg/api/deep_copy_generated.go and /home/mkargaki/go/src/github.com/kargakis/origin/_output/verify-generated-deep-copies/pkg/api/deep_copy_generated.go differ
Files /home/mkargaki/go/src/github.com/kargakis/origin/pkg/api/v1/deep_copy_generated.go and /home/mkargaki/go/src/github.com/kargakis/origin/_output/verify-generated-deep-copies/pkg/api/v1/deep_copy_generated.go differ
Files /home/mkargaki/go/src/github.com/kargakis/origin/pkg/api/v1beta3/deep_copy_generated.go and /home/mkargaki/go/src/github.com/kargakis/origin/_output/verify-generated-deep-copies/pkg/api/v1beta3/deep_copy_generated.go differ
FAILURE: Current deep copies out of date. Please run hack/update-generated-deep-copies.sh

[mkargaki@dhcp129-211 origin]$ ./hack/update-generated-deep-copies.sh 
Generating for version api
Generating for version v1beta3
Generating for version v1

[mkargaki@dhcp129-211 origin]$ ./hack/verify-generated-deep-copies.sh 
===== Verifying Generated Conversions =====
Building gendeepcopy binary...
FAILURE: Building gendeepcopy binary failed:
++ Building go targets for linux/amd64: cmd/gendeepcopy
# github.com/openshift/origin/pkg/api
_output/local/go/src/github.com/openshift/origin/pkg/api/deep_copy_generated.go:191: cannot use nil as type "k8s.io/kubernetes/pkg/api".ObjectReference in assignment
_output/local/go/src/github.com/openshift/origin/pkg/api/deep_copy_generated.go:560: cannot use nil as type "k8s.io/kubernetes/pkg/api".ObjectReference in assignment
_output/local/go/src/github.com/openshift/origin/pkg/api/deep_copy_generated.go:1046: cannot use nil as type "k8s.io/kubernetes/pkg/api".EnvVar in assignment
_output/local/go/src/github.com/openshift/origin/pkg/api/deep_copy_generated.go:1099: cannot use nil as type "k8s.io/kubernetes/pkg/api".EnvVar in assignment
_output/local/go/src/github.com/openshift/origin/pkg/api/deep_copy_generated.go:1180: cannot use nil as type "k8s.io/kubernetes/pkg/api".EnvVar in assignment
_output/local/go/src/github.com/openshift/origin/pkg/api/deep_copy_generated.go:1226: cannot use nil as type "k8s.io/kubernetes/pkg/api".EnvVar in assignment
_output/local/go/src/github.com/openshift/origin/pkg/api/deep_copy_generated.go:1463: cannot use nil as type "k8s.io/kubernetes/pkg/api".EnvVar in assignment

kargakis commented:
@liggitt should we cherry-pick https://github.com/kubernetes/kubernetes/pull/13837 ?

deads2k commented:
> @liggitt should we cherry-pick kubernetes/kubernetes#13837 ?

No.  We turn off `--validate` by default.  If we decide we need it, we'll pull it in after the rebase.

kargakis commented:
> No. We turn off --validate by default. If we decide we need it, we'll pull it in after the rebase.

Now we do: https://github.com/kargakis/origin/commit/efd136920932116484ac9aebeaa548389e21ce55

deads2k commented:
> Now we do: kargakis@efd1369

Start an issue for rebase followups and add one for "update swagger enough for --validate to work well"

kargakis commented:
Squashed @pweil-'s commit into v1beta3 and the custom conversion commit respectively. Test fix got into origin code.

pweil- commented:
@kargakis one more fix for you for the last unit test that was broken.  It seems as if the etcd prefix here was empty and resulting in looking for keys like `//foo` which was blowing up - https://github.com/pweil-/origin/commit/81a968226df6c983d942dd8512bf60c89cb99de2 .  Since I have never used the thirdparty portions I'm not 100% sure this correct so if someone more familiar with that could give it a review...

/cc @deads2k

liggitt commented:
I thought @smarterclayton had already turned off validation... did we drop something, or was that not already done?

deads2k commented:
@kargakis https://github.com/kargakis/origin/pull/6 gets test-cmd and test-e2e passing.  It gets all but TestDNS and TestBuildDeleteController passing for test-integration.

TestDNS is only failing on the local A record, so you might suppress that one test and the Build test (looks like a fragile watch) to see if you get a clean run.

deads2k commented:
@kargakis @pweil- made https://github.com/deads2k/origin/pull/13/files to get the unit tests clean.

pweil- commented:
@kargakis when you merge and re-squash things remove https://github.com/deads2k/origin/pull/13/files.  The better solution is to add `export ETCD_PREFIX="registry"` to `hack/test-go.sh`.  That will cover any future tests that happen to be using the `etcdtest.PathPrefix` call that was failing.  Tested locally, ran clean.

pweil- commented:
@kargakis @deads2k - https://github.com/deads2k/origin/pull/14 should allow a clean `hack/test-integration.sh`.

kargakis commented:
@deads2k @pweil- thanks a lot for your latest PRs. Since this starts wrapping up, can I get initial reviews?

FYI all of the bumps (except for go-restful which is shared) and package removals will be squashed into the Kube bump commit. 

@smarterclayton @ironcladlou @liggitt too

[test]

deads2k commented:
open followup issue for "InitialResources"

deads2k commented:
followup issue for `experimentalMode` on the NamespaceLifecycle since it is so fragile.

deads2k commented:
Followup issue to consider exposing syncPeriod inside `RunProxy`. If there's an existing kube flag, we'll want to plumb it through.

deads2k commented:
Confirm that you have 
```
commit a1d621f5e216283fdd19264968fbcc342fb4ebbd
Author: Clayton Coleman <ccoleman@redhat.com>
Date:   Tue Jun 2 00:19:36 2015 -0400

    UPSTREAM: Hack date-time format on *util.Time
    
    Conflicts:
        Godeps/_workspace/src/github.com/emicklei/go-restful/swagger/model_builder.go

commit 14e1099bb0477a06f1b31ba6345b087ef97f2f4a
Author: Clayton Coleman <ccoleman@redhat.com>
Date:   Tue Jun 2 10:45:17 2015 -0400

    UPSTREAM: Add "Info" to go-restful ApiDecl
    
    Conflicts:
        Godeps/_workspace/src/github.com/emicklei/go-restful/swagger/swagger.go
```

for go-restful

kargakis commented:
@deads2k both are in

deads2k commented:
open followup for `Flag shorthand -t has been deprecated, please use --template instead`.  This will screw `new-app` in particular, since `--template` in new-app means something different.

deads2k commented:
@kargakis two separate jenkins runs got the same error:

```
!!! Error in hack/../test/end-to-end/core.sh:224
	'return 1' exited with status 1
Call stack:
	1: hack/../test/end-to-end/core.sh:224 main(...)
Exiting with status 1
```

https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/5175/console

deads2k commented:
Actually, there might have been two failures:

```
[INFO] Waiting for command to finish: 'oc get events -n node-selector | grep pod-without-node-name | grep failedScheduling'...
[ ERR] Gave up waiting for: 'oc get events -n node-selector | grep pod-without-node-name | grep failedScheduling'

[FAIL] !!!!! Test Failed !!!!
```

kargakis commented:
> open followup for Flag shorthand -t has been deprecated, please use --template instead. This will screw new-app in particular, since --template in new-app means something different.

--template was already used, they just deprecated -t, right?

danwinship commented:
> Followup issue to consider exposing syncPeriod inside RunProxy. If there's an existing kube flag, we'll want to plumb it through.

Note that they're looking at changing the default upstream: https://github.com/kubernetes/kubernetes/pull/14527

kargakis commented:
> Note that they're looking at changing the default upstream: kubernetes/kubernetes#14527

Thanks for this! @deads2k should we update?

deads2k commented:
> Thanks for this! @deads2k should we update?

I'm not worried about it for the rebase unless someone screams that they're totally busted.

pmorie commented:
Add cases in compatibility_test.go for metadata/downward API volumes

liggitt commented:
we should be able to drop 71afb907bfc5a7e4b5cca801d5623fbc3c8da5fe

kargakis commented:
> Add cases in compatibility_test.go for metadata/downward API volumes

I see https://github.com/openshift/origin/commit/7e0b272924cbc7ee93ace032c91671388559929c failing for downward. Fails for `.path` but passes for `.name`. This is bad.

kargakis commented:
> I see 7e0b272 failing for downward. Fails for .path but passes for .name. This is bad.

My bad, hadn't correctly updated the downward api field in the map. Passes now.

liggitt commented:
Our version of convert_api_VolumeSource_To_v1_VolumeSource changed from using the generated version to using a default reflection-based version... there are performance and compatibility implications to that, right?

deads2k commented:
> Our version of convert_api_VolumeSource_To_v1_VolumeSource changed from using the generated version to using a default reflection-based version... there are performance and compatibility implications to that, right?

Performance: yes, compatiblity: I wouldn't think so.  The alternatives are to not be backwards compatible or risk slips every rebase and have to manually modify generated files (the cycle would be: 1 remove custom conversion, 2. generate conversion, 3. harvest generated code, 4 restore conversion, 5 regenerate conversions).  The sub-fields are still generated, so this only has to walk a single type.  The pain is great enough that I'd rather have proof of "it sucks" before accepting it.

liggitt commented:
or have a custom conversion function that calls the generated one first instead of `s.DefaultConvert`

deads2k commented:
> or have a custom conversion function that calls the generated one first instead of s.DefaultConvert, and register it instead of the generated one

The code won't be generated if the custom conversion exists.  Leading to the 5 step loop during rebases.

liggitt commented:
seeing lots of unexpected flags on all commands:
```
+    flags+=("--show-all")
+    flags+=("-a")
+    flags+=("--sort-by=")
```
and lots of test flags appearing where I wouldn't expect them:
```
+    flags+=("--test.bench=")
+    flags+=("--test.benchmem")
+    flags+=("--test.benchtime=")
+    flags+=("--test.blockprofile=")
+    flags+=("--test.blockprofilerate=")
+    flags+=("--test.coverprofile=")
+    flags+=("--test.cpu=")
+    flags+=("--test.cpuprofile=")
+    flags+=("--test.memprofile=")
+    flags+=("--test.memprofilerate=")
+    flags+=("--test.outputdir=")
+    flags+=("--test.parallel=")
+    flags+=("--test.run=")
+    flags+=("--test.short")
+    flags+=("--test.timeout=")
+    flags+=("--test.v")
```
(reported upstream in https://github.com/kubernetes/kubernetes/issues/8744 and supposedly fixed... we may need to do something similar)

liggitt commented:
There's a new port open when running all-in-one:
```
Expected: 53 4001 7001 8443 10250 single-high-port-for-kubelet-api-proxy
Open:     53 4001 7001 8443 10250 37938 55539
```
one port is the kubelet proxy to the apiserver, but the other is unknown. curling it spits out this error in the all-in-one log:
```
E0925 14:17:22.756154   29343 proxysocket.go:104] Dial failed: dial tcp 10.13.129.51:8443: i/o timeout
E0925 14:17:24.756656   29343 proxysocket.go:104] Dial failed: dial tcp 10.13.129.51:8443: i/o timeout
E0925 14:17:28.757025   29343 proxysocket.go:104] Dial failed: dial tcp 10.13.129.51:8443: i/o timeout
E0925 14:17:36.757507   29343 proxysocket.go:104] Dial failed: dial tcp 10.13.129.51:8443: i/o timeout
E0925 14:17:36.757597   29343 proxysocket.go:138] Failed to connect to balancer: failed to connect to an endpoint.
```
we need to figure out what is opening this port and what it is exposing

deads2k commented:
`sort-by` is new and expected.  `show-all` is ugly, but exists upstream.

deads2k commented:
> one port is the kubelet proxy to the apiserver, but the other is unknown. curling it spits out this error in the all-in-one log:

I thought @kargakis fixed up the script in some way.  Did he accidentally break it?

liggitt commented:
all he did was change exit codes... running the version from master gives the same results

liggitt commented:
ok, sweep done

deads2k commented:
> ok, sweep done

@kargakis In case you didn't notice: two items added to your TODOs in the description.

kargakis commented:
> all he did was change exit codes... running the version from master gives the same results

That's strange. I was running the script on Thursday or Friday morning and it was passing for me (no unexpected port) but now -rebased on top of the current master- I see more ports too.
```
[vagrant@openshiftdev origin]$ ./hack/verify-open-ports.sh 
[INFO] Checking open ports ('sudo openshift start' should already be running)
Expected: 53 4001 7001 8443 10250 single-high-port-for-kubelet-api-proxy
Open:     53 4001 7001 8443 10250 36485 39178 44248 44929
```

kargakis commented:
I rerun the Openshift server with fresh etcd and `verify-open-ports.sh` now passes for me
```
[vagrant@openshiftdev origin]$ ./hack/verify-open-ports.sh 
[INFO] Checking open ports ('sudo openshift start' should already be running)
Found expected ports open (53 4001 7001 8443 10250 37977)
```

pweil- commented:
@kargakis The latest failure appears to be related to the etcd_prefix.  It isn't liking the "thirdparty", I used "registry" locally and had clean runs.  I'd say that if that isn't working for you we should probably merge and figure it out in a follow up.

```
<autogenerated>:29: data for /thirdparty/thirdpartyresources/test/foo2 was not defined prior to invoking Get
```

What else is left on this?  If I read correctly, the issue with the new port was a run with etcd that wasn't clean?

kargakis commented:
@pweil- regarding the etcd prefix issue, I think we should probably fallback to the first solution we had for this which was a carry patch. WDYT?

Other than that the only issue that is left on this is the test flags linked to the client binaries, though I cannot find where in our non-testing code we link to the `testing` package (upstream solution: https://github.com/kubernetes/kubernetes/issues/8744). Maybe a follow-up issue for this is enough for now?

smarterclayton commented:
Test flags in client binaries are bad.  Use go list -f '{{ .Deps }}' to
narrow down which packages include testing

On Sep 28, 2015, at 9:00 PM, Michail Kargakis <notifications@github.com>
wrote:

@pweil- <https://github.com/pweil-> regarding the etcd prefix issue, I
think we should probably fallback to the first solution we had for this
which was a carry patch. WDYT?

Other than that the only issue that is left on this is the test flags
linked to the client binaries, though I cannot find where in our
non-testing code we link to the testing package (upstream solution:
kubernetes/kubernetes#8744
<https://github.com/kubernetes/kubernetes/issues/8744>). Maybe a follow-up
issue for this is enough for now?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4663#issuecomment-143844356>.

pweil- commented:
> @pweil- regarding the etcd prefix issue, I think we should probably fallback to the first solution we had for this which was a carry patch. WDYT?

If "registry" doesn't work for you then I'm fine with a follow up.

deads2k commented:
> Test flags in client binaries are bad.  Use go list -f '{{ .Deps }}' to
> narrow down which packages include testing

I agree its bad, is it really bad enough to hold up the rest of the rebase for?  I'd prefer to let it in and fix it as a followup.  It looks ugly, but its not unstable for reasonable usage.

kargakis commented:
Is the test directory excluded from the binary?

pweil- commented:
@kargakis - I think I've tracked this down (thanks to the `go list` advice above).  Please give https://github.com/kargakis/origin/pull/8 a shot and regen completions.

kargakis commented:
OK, only thing left is green e2e:)

kargakis commented:
Jenkins is green! Is Travis failure a flake? http://pastebin.com/9HSHTp7C

kargakis commented:
now another one in integration tests: http://pastebin.com/vSFaMe2L

deads2k commented:
lgtm [merge]  

@liggitt @smarterclayton @pweil- its the in queue, but I think all objects have been covered.
